### PR TITLE
Ports chasm exceptions to typecaches

### PR DESCRIPTION
**Pull request by ghost**
Ya know, for shitcode, spacepods did clean up a LOT of code

### CODE REVIEW

File: code/game/turfs/simulated/chasm.dm [hunk 1]
```diff
@@ -88,12 +89,7 @@
 		return FALSE
 	if(!isliving(AM) && !isobj(AM))
 		return FALSE
-	if(istype(AM, /obj/singularity) || istype(AM, /obj/item/projectile) || AM.throwing)
```
Comment (ChangelingRain) [inline code/game/turfs/simulated/chasm.dm]: missed one

Comment (MoreRobustThanYou) [reply code/game/turfs/simulated/chasm.dm]: fixed

File: code/game/turfs/simulated/chasm.dm [hunk 2]
```diff
@@ -11,6 +11,7 @@
 	canSmoothWith = list(/turf/open/floor/fakepit, /turf/open/chasm)
 	density = TRUE //This will prevent hostile mobs from pathing into chasms, while the canpass override will still let it function like an open turf
 	var/static/list/falling_atoms = list() //Atoms currently falling into the chasm
+	var/static/list/forbidden_types = typecacheof(list(/obj/effect/portal, /obj/singularity, /obj/structure/stone_tile, /obj/item/projectile))
```
Comment (Cyberboss) [inline code/game/turfs/simulated/chasm.dm]: Should be all of obj/effect tbh instead of just portal

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/turfs/simulated/chasm.dm (hunk 2 above)
Comment (Iamgoofball) [reply code/game/turfs/simulated/chasm.dm]: foam should fall down the chasm

and why would smoke fall down it

Comment (MoreRobustThanYou) [reply code/game/turfs/simulated/chasm.dm]: that'd cause like, blood to not fall.

**Review by Cyberboss — DISMISSED** [review]

File: code/game/turfs/simulated/chasm.dm (hunk 2 above)
Comment (Cyberboss) [inline code/game/turfs/simulated/chasm.dm]: At least add obj/effect/abstract and obj/effect/temp_visual

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — DISMISSED** [review]

**Review by Cyberboss — APPROVED** [review]

File: code/game/turfs/simulated/chasm.dm (hunk 2 above)
Comment (MoreRobustThanYou) [reply code/game/turfs/simulated/chasm.dm]: done

**Pull request merged**